factorial/fact.cpp

```diff
@@ -0,0 +1,36 @@
+#include<iostream>
+int fun(int n)
+{
+if(n==0)
+    {
+        return 1;
+    }
+    else
+    {
+            n=n*fun(n-1);
+    }
+}
+int main()
+{
+    int n;
```

Comment:
What is n? Try to give meaningful names to variables. If this variable is supposed to hold input then you can name it as input or userInput. Try to read C++ naming convention from google